driver/pace/driver/run.py

```diff
@@ -116,82 +130,179 @@ class SerialboxConfig(InitializationConfig):
     ) -> DriverState:
-        raise NotImplementedError()
+        (
+            grid,
+            grid_data,
+            damping_coeff,
+            driver_grid_data,
+        ) = self.get_grid_data_damping_coeff_and_driver_grid(
+            quantity_factory, communicator
+        )
+        dycore_state = self._initialize_dycore_state(quantity_factory, communicator)
+        physics_state = self._init_physics_state(quantity_factory)
+        return DriverState(
+            dycore_state=dycore_state,
+            physics_state=physics_state,
+            grid_data=grid_data,
+            damping_coefficients=damping_coeff,
+            driver_grid_data=driver_grid_data,
+        )
 
     def _initialize_dycore_state(
-        self, communicator: pace.util.CubedSphereCommunicator
+        self,
+        quantity_factory: pace.util.QuantityFactory,
+        communicator: pace.util.CubedSphereCommunicator,
     ) -> fv3core.DycoreState:
-        # TODO: this code currently depends on namelist and stencil_factory,
-        # update the config to require a namelist file or load the file from
-        # the place we know it exists in
-        # the file path, and use that to make a factory.
-        #
-        # import sys
-        # sys.path.append("/usr/local/serialbox/python/")
-        # import serialbox
-
-        # serializer = serialbox.Serializer(
-        #     serialbox.OpenModeKind.Read,
-        #     self._data_dir,
-        #     "Generator_rank" + str(self._comm.rank),
-        # )
-
-        # grid = fv3core._config.make_grid(
-        #     npx=self.npx,
-        #     npy=self.npy,
-        #     npz=self.npz,
-        #     layout=self.layout,
-        #     rank=communicator.rank,
-        #     backend=self.backend,
-        # )
-        # savepoint_in = serializer.get_savepoint("FVDynamics-In")[0]
-        # translate_object = TranslateFVDynamics([grid], namelist, stencil_factory)
-        # input_data = translate_object.collect_input_data(serializer, savepoint_in)
-        # dycore_state = translate_object.state_from_inputs(input_data)
-        # return dycore_state
-        pass
-
-    def _init_physics_state_from_dycore_state(
+        (
+            grid,
+            grid_data,
+            damping_coeff,
+            driver_grid_data,
+        ) = self.get_grid_data_damping_coeff_and_driver_grid(
+            quantity_factory, communicator
+        )
+        ser = self.serializer(communicator)
+        savepoint_in = ser.get_savepoint("FVDynamics-In")[0]
+        stencil_factory = StencilFactory(
+            config=self.stencil_config, grid_indexing=grid.grid_indexing
+        )
+        translate_object = TranslateFVDynamics([grid], self.namelist, stencil_factory)
+        input_data = translate_object.collect_input_data(ser, savepoint_in)
+        dycore_state = translate_object.state_from_inputs(input_data)
+        return dycore_state
+
+    def _init_physics_state(
```

Comment:
You can delete this method and call the code below directly where you'd call this function, since the call signatures are similar size and the code contained here would be more explicit.